Sources/ApodiniNetworking/HTTPServer.swift

```diff
@@ -0,0 +1,402 @@
+            eventLoopGroup: eventLoopGroup,
+            tlsConfiguration: tlsConfiguration,
+            enableHTTP2: enableHTTP2,
+            address: address,
+            logger: logger
+        ))
+        self.router = HTTPRouter(logger: logger)
+    }
+    
+    
+    deinit {
+        switch config {
+        case .app:
+            break
+        case .custom(let configStorage):
+            switch configStorage.eventLoopGroupProvider {
+            case .shared:
+                break
+            case .createNew:
+                try! self.eventLoopGroup.syncShutdownGracefully()
+            }
+        }
+    }
+    
+    
+    /// Start the server.
+    /// This will attempt to open a NIO channel, bind it to the specified address, and set it up to handle incoming HTTP and HTTP2 requests, depending on the configuration options.
+    public func start() throws {
+        guard channel == nil else {
+            throw ApodiniNetworkingError(message: "Cannot start already-running servers")
+        }
+        let bootstrap = ServerBootstrap(group: eventLoopGroup)
+            .serverChannelOption(ChannelOptions.backlog, value: 256)
+            .serverChannelOption(ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEADDR), value: 1)
+            .childChannelInitializer { [unowned self] (channel: Channel) -> EventLoopFuture<Void> in
+                logger.notice("Configuring NIO channel pipeline. TLS: \(tlsConfiguration != nil), HTTP/2: \(enableHTTP2)")
+                if let tlsConfig = tlsConfiguration {
+                    precondition(tlsConfig.applicationProtocols.contains("h2"), "h2 not found in \(tlsConfig.applicationProtocols)")
+                    let sslContext: NIOSSLContext
+                    do {
+                        sslContext = try NIOSSLContext(configuration: tlsConfig)
+                    } catch {
+                        logger.error("Unable to configure TLS: \(error)")
+                        return channel.close(mode: .all)
+                    }
+                    let tlsHandler = NIOSSLServerHandler(context: sslContext)
+                    return channel.pipeline.addHandler(tlsHandler)
+                        .flatMap { () -> EventLoopFuture<Void> in
+                            return channel.configureHTTP2SecureUpgrade { channel in
+                                channel.addApodiniNetworkingHTTP2Handlers(responder: self)
+                            } http1ChannelConfigurator: { channel in
+                                channel.addApodiniNetworkingHTTP1Handlers(responder: self)
+                            }
+                        }
+                        .flatMapError { error in
+                            channel.eventLoop.makeFailedFuture(error)
+                        }
+                } else {
+                    if enableHTTP2 {
+                        // NOTE this doesn't make sense and (probably) doesn't work
```

Comment:
May this be an outdated comment? Otherwise, why would it not work? Will this be tested? Will this be resolved in this PR? Otherwise I would suggest to track this with an Issue providing a bit more detail about the problem situation.

Reply:
Nope, very much not outdated. Basically no one supports plaintext HTTP/2 (which is what you'd end up configuring in that branch). Technically speaking, plaintext h2 is a thing, but there seems to be a consensus that everyone refuses to implement it (see eg also Nginx and (iirc) Chrome).

[Previously, this would've been a crash](https://github.com/vapor/vapor/blob/a8bd13fa10b14d2c014f002a4eed0558f66bcf6f/Sources/Vapor/HTTP/Server/HTTPServer.swift#L369). If you think that's a more appropriate response we can also add a fatal error here, I'd be up for that.